Backpack/BarChart/Classes/BarChartBar.swift

```diff
@@ -0,0 +1,181 @@
+            self.barView.widthAnchor.constraint(equalToConstant: BPKSpacingBase),
+            self.barView.bottomAnchor.constraint(equalTo: self.backgroundView.bottomAnchor),
+            self.barView.centerXAnchor.constraint(equalTo: self.centerXAnchor),
+            self.backgroundView.topAnchor.constraint(equalTo: self.topAnchor),
+            self.backgroundView.centerXAnchor.constraint(equalTo: self.centerXAnchor),
+            self.miniFlareView.centerXAnchor.constraint(equalTo: self.backgroundView.centerXAnchor),
+            self.titleLabel.centerXAnchor.constraint(equalTo: self.backgroundView.centerXAnchor),
+            self.subtitleLabel.centerXAnchor.constraint(equalTo: self.backgroundView.centerXAnchor),
+            self.titleLabel.topAnchor.constraint(equalTo: self.backgroundView.bottomAnchor, constant: BPKSpacingMd),
+            self.subtitleLabel.topAnchor.constraint(equalTo: self.titleLabel.bottomAnchor, constant: BPKSpacingMd),
+            self.bottomAnchor.constraint(equalTo: self.subtitleLabel.bottomAnchor, constant: BPKSpacingMd)
+        ])
+        self.updateBarHeight()
+        self.updateminiFlareViewAppearance()
+    }
+
+    func updateBarColor() {
+        if self.isSelected {
+            barView.backgroundColor = self.selectedColor
+            titleLabel.textColor = self.selectedColor
+            subtitleLabel.textColor = self.selectedColor
+            return
+        }
+
+        titleLabel.textColor = BPKColor.textPrimaryColor
+        subtitleLabel.textColor = BPKColor.textPrimaryColor
+
+        if self.privateFillValue == nil {
+            barView.backgroundColor = self.noPriceColor
+        } else {
+            barView.backgroundColor = self.priceColor
+        }
+    }
+
+    func updateBarHeight() {
+        if self.barViewHeightAnchor != nil {
+            self.barViewHeightAnchor!.isActive = false
+        }
+
+        if self.privateFillValue == nil {
+            self.barViewHeightAnchor = self.barView.heightAnchor.constraint(equalToConstant: BPKSpacingLg)
+        } else {
+            self.barViewHeightAnchor = self.barView.heightAnchor.constraint(equalTo: self.backgroundView.heightAnchor,
+                                                                            multiplier: self.privateFillValue!)
+        }
+
+        self.barViewHeightAnchor!.isActive = true
+    }
+
+    func updateminiFlareViewAppearance() {
+        if self.miniFlareViewPositionAnchor != nil {
+            self.miniFlareViewPositionAnchor!.isActive = false
+        }
+
+        self.miniFlareViewPositionAnchor = self.miniFlareView.bottomAnchor.constraint(equalTo: self.barView.topAnchor,
+                                                                                      constant: -BPKSpacingSm)
+
+        self.miniFlareViewPositionAnchor!.isActive = true
+
+        self.miniFlareView.isHidden = self.realValue == nil || !self.isSelected
+    }
+
+    public override var isSelected: Bool {
```

Comment:
Can we keep public variables at the top of the file? These also need to be documented